### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by davidask**
Related to [SQL Refactor](https://github.com/Zewo/SQL/pull/13).

### CODE REVIEW

File: Package.swift [hunk 1]
```diff
@@ -4,6 +4,6 @@ let package = Package(
 	name: "PostgreSQL",
 	dependencies: [
 		.Package(url: "https://github.com/Zewo/CLibpq.git", majorVersion: 0, minor: 5),
-		.Package(url: "https://github.com/Zewo/SQL.git", majorVersion: 0, minor: 8),
+		.Package(url: "https://github.com/Zewo/SQL.git", majorVersion: 0, minor: 0)
```
Comment (Danappelxx) [inline Package.swift:7]: before merging this make sure its `minor: 9`

Comment (davidask) [inline Package.swift:7]: Yeah, this is for the developing tag.

File: Sources/QueryRenderer.swift [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+        var components = ["DELETE FROM", statement.tableName]
+        
+        if let predicate = statement.predicate {
+            components.append("WHERE")
+            components.append(composePredicate(predicate))
+        }
+        
+        return components.joined(separator: " ")
+    }
+}
```
Comment (Danappelxx) [inline Sources/QueryRenderer.swift]: line break :)

File: Sources/FieldInfo.swift [hunk 3]
```diff
@@ -25,9 +25,11 @@
...
-    public var name: String
+    public let name: String
+    public let index: Int
     
-    init(name: String) {
+    init(name: String, index: Int) {
         self.name = name
+        self.index = index
     }
 }
```
Comment (Danappelxx) [inline Sources/FieldInfo.swift]: here too

File: Sources/Row.swift [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    public let result: Result
+    
+    public let index: Int
+    
+    public init(result: Result, index: Int) {
+        self.result = result
+        self.index = index
+    }
+}
+
```
Comment (Danappelxx) [inline Sources/Row.swift]: one too many here

File: Sources/Result.swift [hunk 5]
```diff
@@ -145,28 +136,21 @@ public class Result: SQL.ResultProtocol {
...
         for i in 0..<PQnfields(self.resultPointer) {
             guard let fieldName = String(validatingUTF8: PQfname(self.resultPointer, i)) else {
                 continue
             }
             
-            result.append(
-                FieldInfo(name: fieldName)
-            )
+            let ok = PQftype(self.resultPointer, i)
+            print(ok)
```
Comment (Danappelxx) [inline Sources/Result.swift]: print should probably not be here

**Comment by Danappelxx** [comment]
Very late but excellent work nevertheless 💯

**Pull request merged**